kview/src/main/java/dev/ikm/komet/kview/events/ShowPatternPanelEvent.java

```diff
@@ -51,4 +51,27 @@ public class ShowPatternPanelEvent extends Evt {
     public ShowPatternPanelEvent(Object source, EvtType eventType) {
         super(source, eventType);
     }
+
+    /**
+     * Constructs a prototypical Event.
+     *
+     * @param source    the object on which the Event initially occurred
+     * @param eventType
+     * @param fieldOrders the total number of field orders in the current pattern.
+     * @throws IllegalArgumentException if source is null
+     */
+    public ShowPatternPanelEvent(Object source, EvtType eventType, int fieldOrders) {
+        super(source, eventType);
+        this.fieldOrders = fieldOrders;
+    }
+
+    private int fieldOrders;
```

Comment:
Can you make this `final`?

Reply:
done, changed the variable name to "fields" and made it final. Keeping it plural since its the number of fields in the details section could be 0,1 or more.